src/key.rs

```diff
@@ -418,6 +443,19 @@ impl PublicKey {
         }
     }
 
+    /// Creates a [`PublicKey`] using the key material from `xonly` combined with `parity`.
+    pub fn from_x_only_public_key_and_parity(xonly: XOnlyPublicKey, parity: Parity) -> Result<PublicKey, Error> {
```

Comment:
Rust API guidelines recommend to name parameters taking many args to use `with_*` pattern. I propose here `with_xpubkey_parity`

Reply:
I think `from` is clearer because we're literally transforming the inputs, the parameters are not modifiers on a default `PublicKey`.

Reply:
I went looking and couldn't find the API docs for with vs for, if you disagree with @apoelstra could you please post a link @dr-orlovsky?

Re the ridiculously long name; I was trying to be uniform with the other functions that use `x_only_public_key` in the function name. However `x_only_public_key` functions also return the parity now so possibly we could shorten them all to `xonly`, users would know that 'xonly' means `XOnlyPublicKey` and `Parity`?

As an example

```rust
    /// Returns the [`XOnlyPublicKey`] (and it's [`Parity`]) for this [`SecretKey`].
    ///
    /// This is equivalent to `XOnlyPublicKey::from_keypair(self.keypair(secp))`.
    #[inline]
    pub fn xonly<C: Signing>(&self, secp: &Secp256k1<C>) -> Result<(XOnlyPublicKey, Parity), Error> {
        let kp = self.keypair(secp);
        XOnlyPublicKey::from_keypair(&kp)
    }
```

Reply:
@tcharding https://github.com/rust-lang/rfcs/blob/master/text/0430-finalizing-naming-conventions.md#general-naming-conventions

I had read in some previous docs on rust naming in APIs that `with_*` stands for when we have more than a one parameter, while `from` are the conversions from a specific single type.

I will agree with any naming: all my naming comments are just nits; in fact with modern IDE I do not really care about function names since I always check the function content when do not remember what it does.

Reply:
`from_xonly` is good by me. Have a mild preference for `from_xonly_parity` but it's fine, the fact that you need the parity is clear from the function signature (and will be clear from the compiler error if you use it wrong).